sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,173 @@
+        """
+
+        def _match_namespace(namespace):
+            """Match a particular namespace for inclusion (or not) in the
+            collection phases
+
+                :param namespace str:   The name of a namespace
+            """
+
+            for regex in self.collect_regexes:
+                if re.match(regex, namespace):
+                    return True
+            return False
+
+        self.collect_regexes = self._setup_namespace_regexes()
+
+        return list(set([n for n in nsps if _match_namespace(n)]))
+    
+    def _get_namespaces(self):
+        res = self.exec_cmd('microshift get namespaces -o custom-columns=NAME:.metadata.name --no-headers')
+        if res['status'] == 0:
+            return self._reduce_namespace_list(res['output'].split('\n'))
+        return []
+
+    def _collect_cluster_resources(self):
+        """Collect cluster-level (non-namespaced) resources
+        """
+        global_resources = [
+            'apiservices',
+            'certificatesigningrequests',
+            'clusterrolebindings',
+            'clusterroles',
+            'componentstatuses',
+            'csidrivers',
+            'csinodes',
+            'customresourcedefinitions',
+            'flowschemas',
+            'ingressclasses',
+            'logicalvolumes',
+            'mutatingwebhookconfigurations',
+            'nodes',
+            'persistentvolumes',
+            'priorityclasses',
+            'prioritylevelconfigurations',
+            'rangeallocations',
+            'runtimeclasses',
+            'securitycontextconstraints',
+            'selfsubjectaccessreviews',
+            'selfsubjectrulesreviews',
+            'storageclasses',
+            'subjectaccessreviews',
+            'tokenreviews',
+            'validatingwebhookconfigurations',
+            'volumeattachments'
+        ]
+
+        _filtered_resources = []
+
+        # Filter out any resources that might not exist.
+        for resource in global_resources:
+            res = self.exec_cmd("microshift get %s" % resource)
+            if res['status'] == 0:
+                _filtered_resources.append(resource)
+
+        _subdir = self.get_cmd_output_path(make=False)
+        self.exec_cmd(
+            'microshift inspect --kubeconfig %s --dest-dir %s %s' % (
+                self.get_option('kubeconfig'),
+                _subdir,
+                ",".join(_filtered_resources)))
+
+    def _collect_named_resources(self):
+        """Collect namespaced resources (including pod logs) from designated
+        namespaces
+        """
+        _namespaces = ['ns/%s' % n for n in self._get_namespaces()]
+        _subdir = self.get_cmd_output_path(make=False)
+        res = self.exec_cmd(
+            'microshift inspect --kubeconfig %s --dest-dir %s %s' % (
+                self.get_option('kubeconfig'),
+                _subdir,
+                " ".join(_namespaces)))
```

Comment:
Same here as above - either call `exec_cmd()` during `collect()`, or change this to `add_cmd_output()` with a particular `suggest_filename` so that we get both the resulting output file from `microshift inspect` and any potential error output from the command.

Reply:
changed it to `add_cmd_output`